Policyfile.rb

```diff
@@ -0,0 +1,13 @@
+name 'haproxy'
+
+default_source :supermarket, 'https://supermarket.chef.io'
+cookbook 'haproxy', path: '.'
+cookbook 'test', path: 'test/cookbooks/test'
+
+run_list 'haproxy'
+
+tests = (Dir.entries('./test/cookbooks/test/recipes').select { |f| !File.directory? f })
```

Comment:
Can be replaced with ``Dir.each_child('./test/cookbooks/test/recipes') { |test| named_run_list test.to_sym, "test:#{test}" }``

Reply:
Yes it can replace the lot, I'm not sure if there's a way to review multiple lines but that's what I meant.

Reply:
Yes it can replace the lot, I'm not sure if there's a way to review multiple lines but that's what I meant.

Reply:
Cool! testing it locally on one of my cookbooks. i like a 1 liner much better for this

Reply:
I didn't find a nicer way to strip .rb off the end of the file so the test = test.gsub is still there

Reply:
`test` -> `File.basename(test, '.*')`

Thought I'd put that in there but apparently not.

Reply:
```ruby
Dir.each_child('./test/fixtures/cookbooks/test/recipes') { |t| named_run_list File.basename(t, .*).to_sym, "test:#{File.basename(t, .*)}" }
```

Reply:
finally got it working. was missing a few `:` tested in my other cookbooks
```ruby
Dir.each_child('./test/cookbooks/test/recipes') { |t| named_run_list :"#{File.basename(t, '.*')}".to_sym, "test::#{File.basename(t, '.*')}" }
```

Reply:
That's curious (assuming named_run_list is a method and not an attribute) as basename returns a string so the string interpolation should be superfluous.

Reply:
the `named_run_list` name here wouldn't be a string though. it is more like creating a property in a cookbook resource. in that sense, `:this_recipe` is then consumed inside the `kitchen.yml` as `named_run_list: this_recipe`